code/modules/pda/PDA.dm

```diff
@@ -165,15 +165,16 @@ GLOBAL_LIST_EMPTY(PDAs)
 		to_chat(usr, "<span class='notice'>You cannot do this while restrained.</span>")
 
 /obj/item/pda/AltClick(mob/user)
-	..()
+	if(!HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) && ((src in user) || (isturf(loc) && in_range(src, user))))
```

Comment:
I believe in_range will let you use this through stuff like windoors/windows. I think you'll want adjacent in this situation
```suggestion
	if(!HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) && ((src in user) || (isturf(loc) && Adjacent(src, user))))
```

Reply:
man in_range is just a weird proc because it just doesn't really work sometimes